pymc_marketing/clv/models/basic.py

```diff
@@ -173,25 +146,28 @@ def load(cls, fname: str):
         """
         filepath = Path(str(fname))
         idata = az.from_netcdf(filepath)
-        dataset = idata.fit_data.to_dataframe()
+        return cls._build_with_idata(idata)
 
+    @classmethod
+    def _build_with_idata(cls, idata: az.InferenceData):
+        dataset = idata.fit_data.to_dataframe()
         model = cls(
             dataset,
             model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
             sampler_config=json.loads(idata.attrs["sampler_config"]),
         )
         model.idata = idata
-
         model.build_model()  # type: ignore
-
         if model.id != idata.attrs["id"]:
-            raise ValueError(
-                f"The file '{fname}' does not contain an inference data of the same model or configuration as '{cls._model_type}'"
-            )
-        # All previously used data is in idata.
-
+            raise ValueError(f"Inference data not compatible with {cls._model_type}")
         return model
 
+    def thin_fit_result(self, keep_every: int):
+        self.fit_result  # Raise Error if fit didn't happen yet
+        assert self.idata is not None
```

Comment:
Same as above, replace this `assert` with an IF statement raising a `RunTimeError`.

Reply:
Same